### PR TITLE
Implement missing Pathname methods

### PR DESCRIPTION
**Pull request by Mogztter**
- split
 - dirname
 - basename
 - directory?
 - extname
 - comparaison

### CODE REVIEW

File: stdlib/pathname.rb [hunk 1]
```diff
@@ -66,6 +73,33 @@ def join(*args)
...
+  def directory?
+    File.directory?(@path)
+  end
+
+  def extname
+    File.extname(@path)
+  end
+
+  def <=>(other)
+    `self.path > other.path ? 1 : (self.path < other.path ? -1 : 0)`
```
Comment (IlyaBylich) [inline stdlib/pathname.rb]: I'm pretty sure it can be replaced with `self.path <=> other.path`

Comment (Mogztter) [inline stdlib/pathname.rb]: @iliabylich Thanks, I will try :+1:

File: stdlib/pathname.rb [hunk 2]
```diff
@@ -1,7 +1,14 @@
+require 'corelib/comparable'
+
 class Pathname
+  include Comparable
+
   def initialize(path)
-    raise ArgumentError if path == "\0"
```
Comment (elia) [inline stdlib/pathname.rb]: you should probably retain this line

File: stdlib/pathname.rb [hunk 3]
```diff
@@ -37,7 +44,7 @@ def cleanpath
   end
 
   def to_path
-    @path
+    @path.to_s
```
Comment (elia) [inline stdlib/pathname.rb]: what about doing this in the initialize, I think many methods below assume it's a string anyway

File: stdlib/pathname.rb (hunk 2 above)
Comment (Mogztter) [inline stdlib/pathname.rb]: could you please explain the purpose of this line ? I don't remember exactly but this cause me troubles :neutral_face:

File: stdlib/pathname.rb (hunk 3 above)
Comment (Mogztter) [inline stdlib/pathname.rb]: good idea, I'm just a little bit concerned with `<=>` method but I think it's better to do it in the initialize for consistency.

**Comment by elia** [comment]
:+1:

**Pull request merged**